lib/pg.provider.js

```diff
@@ -0,0 +1,286 @@
+'use strict';
+
+const path = require('path');
+
+const metaschema = require('metaschema');
+const pg = require('pg');
+
+const { GSError, codes: errorCodes } = require('./errors');
+const { StorageProvider } = require('./provider');
+const {
+  generateQueryParams,
+  escapeIdentifier,
+  buildWhere,
+} = require('./pg.utils');
+const { PostgresCursor } = require('./pg.cursor');
+
+const recreateIdTrigger = Symbol('recreateIdTrigger');
+
+class PostgresProvider extends StorageProvider {
+  // Create PostgresProvider
+  //   gs - globalstorage instance
+  constructor(gs) {
+    super(gs);
+
+    this.pool = null;
+  }
+
+  // Open PostgresProvider
+  //   options - object, passed to pg
+  //   callback - function(err, StorageProvider)
+  open(options, callback) {
+    metaschema.load(path.join(__dirname, '../schemas/system'), err => {
```

Comment:
IMHO, schemas should already be loaded, when someone opens a gs connection.
Moreover, I'd prefer for metaschema to return a schema object, instead of saving info to global state, in that case, schema should be passed as an argument to the constructor, but that a matter of discussion with @tshemsedinov.

Reply:
Explicit +1, even though this will need to be changed in quite a few places I think this will lead to a much better code when dealing with metaschema (you won't need to load the schemas every time because you don't know if they were loaded yet, the actual loading won't be present in 'uses' classes such as this one). Also, IMO this is an essential part of decoupling metaschema from fs. Furthermore, it will be easier to test as global state hinders the testing quite a lot.

Reply:
+1 We need issue: "Do not store state inside metaschema, return instance with state from `load`"